stdlib/list.gr

```diff
@@ -2,6 +2,21 @@
 
 export *
 
+let init = (length, fn) => {
+  if (length <= 0) {
+    []
+  } else {
+    // TODO: How can I make this better?
```

Comment:
Init for lists is annoying because there's a tradeoff on speed vs. not blowing the stack, i.e. do you just implement it the way you did here, or do you build a list and then reverse it at the end. OCaml picks a threshold and then chooses an implementation based on how many items you want: https://github.com/ocaml/ocaml/blob/03839754f46319aa36d9dad56940a6f3c3bcb48a/stdlib/list.ml#L79

Wasm (or maybe WASI would do this?) doesn't yet let you ask questions like "how big is the call stack?", so we can probably just pick something arbitrary, but fairly conservative, like a couple hundred. I think I actually ran a test a little while back to see how many stack frames you get in different browsers, and I think the lowest was something like 600. We could probably do some research.

Alternatively, if we just document that this is not tail-recursive, that's perfectly fine too 👍

Reply:
Updated with a note!